Sources/MessagingInApp/Gist/Gist.swift

```diff
@@ -37,16 +36,21 @@ class Gist: GistProvider {
         inAppMessageManager: InAppMessageManager,
         queueManager: QueueManager,
         threadUtil: ThreadUtil,
-        sseConnectionManager: SseConnectionManager
+        sseLifecycleManager: SseLifecycleManager
     ) {
         self.logger = logger
         self.gistDelegate = gistDelegate
         self.inAppMessageManager = inAppMessageManager
         self.queueManager = queueManager
         self.threadUtil = threadUtil
-        self.sseConnectionManager = sseConnectionManager
+        self.sseLifecycleManager = sseLifecycleManager
 
         subscribeToInAppMessageState()
+
+        // Start the SSE lifecycle manager to observe app foreground/background events
+        Task {
+            await sseLifecycleManager.start()
```

Comment:
We are  starting `sseLifecycleManager.start()` in init, but there’s no corresponding “stop” API and deinit doesn’t/can’t unsubscribe from store. `CioSseLifecycleManager.deinit` seems to be removeing `NotificationCenter` observers, but store subscription is left to “weak cleanup”?